### PR TITLE
fix issues with send max tokens

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

File: packages/core/lib/services/substrate/service.js [hunk 1]
```diff
@@ -46,7 +46,8 @@ export class SubstrateService {
   async getFeeAmount(params: TransactionParams) {
     validation.getFeeAmount(params);
 
-    const {toAddress, fromAddress, amount} = params;
+    const {toAddress, fromAddress} = params;
+    const amount = getPlainDockAmount(params.amount).toNumber();
```
Comment (mike-parkhill) [inline packages/core/lib/services/substrate/service.js:50]: any chance `getPlainDockAmount()` could return something that can't be converted to a number? And if so, what's the result? Does it throw an exception? Or return 0 or -1 or something?

**Review by mike-parkhill — COMMENTED** [review]
are you able to unit test this with different scenarios to make sure they all still work?

**Comment by maycon-mello** [comment]
> are you able to unit test this with different scenarios to make sure they all still work?

It throws on polkadot-js api, and we are still mocking the api connection. So its not possible to cover this scenario properly

**Comment by maycon-mello** [comment]
Related to this jira https://dock-team.atlassian.net/browse/DCKA-963 @mike-parkhill

File: packages/core/lib/services/substrate/service.js (hunk 1 above)
Comment (maycon-mello) [reply packages/core/lib/services/substrate/service.js:50]: the `validation.getFeeAmount(params);` will throw in case the amount is not a number

Comment (maycon-mello) [reply packages/core/lib/services/substrate/service.js:50]: There is a check for `isNumberValid` on it

Comment (mike-parkhill) [reply packages/core/lib/services/substrate/service.js:50]: are we handling that exception gracefully in the UI already?

Comment (maycon-mello) [reply packages/core/lib/services/substrate/service.js:50]: Thanks, i've added this to dock-app repo https://github.com/docknetwork/dock-app/pull/372

**Review by mike-parkhill — APPROVED** [review]

**Pull request merged**